sale_delivery_date/models/res_partner.py

```diff
@@ -79,13 +97,24 @@ def get_next_windows_start_datetime(self, from_datetime, to_datetime):
 
         Note result can include a start datetime that is before from_datetime
         on the from_datetime weekday
+        Note 2 time windows are in the context of customer's tz.
+        We have to localize from_date, to_datetime first, and return a naive
+        UTC datetime.
 
         :param from_datetime: Datetime object
         :param to_datetime: Datetime object
         :return: List of Datetime objects
         """
         res = list()
-        for this_datetime in date_range(from_datetime, to_datetime, timedelta(days=1)):
+        tz_string = self.tz or self.env.company.partner_id.tz or "UTC"
+        tz = timezone(tz_string)
+        from_datetime_utc = utc.localize(from_datetime)
+        from_datetime_tz = from_datetime_utc.astimezone(tz)
+        to_datetime_utc = utc.localize(to_datetime)
+        to_datetime_tz = to_datetime_utc.astimezone(tz)
+        for this_datetime in date_range(
+            from_datetime_tz, to_datetime_tz, timedelta(days=1)
+        ):
```

Comment:
This method could be simplified as we just need to find the first day for which a time window is defined. 
We don't care on the time.